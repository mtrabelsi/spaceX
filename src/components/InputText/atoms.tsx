import React from 'react'
import s from 'styled-components'
import {
    primarytColor,
    secondaryColor,
    lightColor,
    primaryFontSize,
    fontFamily
  } from './../Global'

export const InputWrap = s.div`
  border-bottom: 1px solid ${lightColor};
  display: flex;
`
export const Input = s.input`
  color: ${lightColor};
  background-color: transparent;
  padding: 5px 0;
  border: none;
  outline: none;
  width: 100%;
  font-size: ${primaryFontSize};
  font-family: ${fontFamily};
  ::placeholder {
    color: ${lightColor};
    font-family: ${fontFamily};
    opacity: 0.4;
  }
`
export const Img = s.img`
  height: 20px;
  padding: 9px;
  cursor: pointer;
`