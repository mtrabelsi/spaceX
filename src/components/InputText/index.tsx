import React, { Component } from 'react';
import { Input, InputWrap, Img } from './atoms';

const InputBasic : React.FC<any> = (props) => {
    return(<InputWrap 
            {...props} 
            style={props.inputWrapperStyle}
        >
        {props.hasIconLeft && 
            props.leftIcon && 
            <Img 
                style={props.leftIconStyle} 
                src={props.leftIcon} 
                onClick={props.leftIconClickHandler} 
            />
        }

        <Input 
            {...props} 
            style={props.inputStyle} 
            type="text" 
            placeholder={props.placeholder} 
        />

        {props.hasIconRight && 
            props.rightIcon && 
            <Img 
                style={props.rightIconStyle} 
                src={props.rightIcon} 
                onClick={props.rightIconClickHandler}  
            />
        }
    </InputWrap>)
}

export default InputBasic