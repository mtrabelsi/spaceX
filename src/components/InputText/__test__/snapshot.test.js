import React from 'react';
import renderer from 'react-test-renderer';
import InputText from '..';

it('Should consider all props and generate unique html based on that', () => {
  const tree = renderer
    .create(<InputText
      inputWrapperStyle={{ color: 'red', padding: 5 }}
      containerStyle={{ width: 200 }}
      value="some text in the input"
      hasIconLeft
      leftIcon="s"
      leftIconClickHandler={() => console.log('im snpashotted left click')}
      rightIconClickHandler={() => console.log('im snpashotted right click')}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
