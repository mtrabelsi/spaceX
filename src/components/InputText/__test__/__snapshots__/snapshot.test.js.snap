// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should consider all props and generate unique html based on that 1`] = `
<div
  className="sc-AxjAm faPnbs"
  style={
    Object {
      "color": "red",
      "padding": 5,
    }
  }
>
  <img
    className="sc-AxiKw cuzgab"
    onClick={[Function]}
    src="s"
  />
  <input
    className="sc-AxirZ eIeUVT"
    type="text"
    value="some text in the input"
  />
</div>
`;
